schemas/logs/salesforce/logout.yml

```diff
@@ -0,0 +1,193 @@
+schema: Salesforce.Logout
+version: 0
+referenceURL: https://developer.salesforce.com/docs/atlas.en-us.api.meta/api/sforce_api_objects_eventlogfile_logout.htm
+description: "Logout events contain details of user logouts."
+parser:
+  csv:
+    delimiter: ","
+    hasHeader: true
+fields:
+  - name: EVENT_TYPE
+    required: true
+    type: string
+    validate:
+      allow: ["Logout"]
+  - name: TIMESTAMP
+    required: true
+    type: string
```

Comment:
should be of type timestamp

Reply:
I gave a try defining a `strftime` pattern:

```yaml
  type: timestamp
  timeFormat: "%Y%m%d%H%M%S.%f"
```

Given `"20210217135741.945"` as input, the output field was not formatted as RFC3339 though:

```
"TIMESTAMP": "20210217135741.945000",
```

Maybe we should stay with float?

Reply:
Hmmm that seems weird. Maybe the timeformat is not working as expected? Or maybe the tool just doesn't print the proper value? How does that compare to the other fields you have declared as timestamp? 

> Maybe we should stay with float?

Fields containing timestamp information should be of type timestamp for a few reasons: 
* In AWS Athena, the field would be declared as `TIMESTAMP` type, then whoever is querying the data can use `timestamp` functions: https://prestodb.io/docs/current/functions/datetime.html to run operations on it (change timezone, compare with other timestamp fileds)
* People reading log files are used to seeing epoch time. Data engineers/DB admins are used to just dealing with `TIMESTAMP` columns. 
* As it stands now, we have two  "TIMESTAMP*" fields in the schema where one of them is `TIMESTAMP` format and another `FLOAT`. For someone not so familiar with the schema that just wants to get some data out, they shouldn't have to think which field is fine to use for their queries and which not. It adds some (minimal) cognitive complexity that is not really needed. If someone wants to convert a `TIMESTAMP` field to epoch for some reason, they can use the appropriate function by Athena. 

Normalizing timestamps is one of the few cases where our ETL is actually manipulating the input data.

Reply:
> Maybe the timeformat is not working as expected? Or maybe the tool just doesn't print the proper value?

I think it does parse the value, but I'll check why it is not converted or displayed correctly.

> How does that compare to the other fields you have declared as timestamp?

`TIMESTAMP_DERIVED` is parsed correctly.

> Fields containing timestamp information should be of type timestamp for a few reasons:

I won't argue on the value of this and it may be indeed confusing to have different types, but shouldn't we always rely on `p_event_time` instead? That should alleviate the cognitive overhead that you mention. Also, I am not sure why the provide both formats, the value is the same.

Reply:
> > Maybe the timeformat is not working as expected? Or maybe the tool just doesn't print the proper value?
> 
> I think it does parse the value, but I'll check why it is not converted or displayed correctly.
> 
> > How does that compare to the other fields you have declared as timestamp?
> 
> `TIMESTAMP_DERIVED` is parsed correctly.
> 
> > Fields containing timestamp information should be of type timestamp for a few reasons:
> 
>but shouldn't we always rely on `p_event_time` instead? 

`p_event_time` is used to define when the event happened so yes, in this case, they could just go to that and ignore the rest. 
Other logs can include more than just one timestamp e.g. [Sophos](https://github.com/panther-labs/panther-analysis/blob/master/schemas/logs/sophos/central.yml#L53-L56),   [Crowdstrike](https://github.com/panther-labs/panther-analysis/blob/master/schemas/logs/crowdstrike/aid_master.yml#L73-L76), etc. In those cases, we "have" to convert the format to `timestamp`... so people eventually get used to just dealing with `TIMESTAMP` type of fields.  

> Also, I am not sure why the provide both formats, the value is the same.
My guesses based on other usecases I've encountered in the past: 
* They found out that integrating with a 3rd party tool is problematic because it expects timestamps in specific format. They added fields retroactively
* The realized that some of their log files has their timestamp in one format while other log files in a different format. People realized it is impossible to do meaningful correlations between events. Opted for a new "standard" which was added. They didn't remove old fields for backwards compatibility.

Reply:
The format is fine.
The output format is the same as the input for all tests.
The override to a common format for all timestamps happens only when writting to S3 using `ConfigForDataLakeWriters`.
This is by design since testing using the input value is less error prone than manually converting between formats.
If the format could not parse the value an error would have occured.
What you see in the output is the value after it was parsed using the `timeFormat` and serialized again back to JSON.

Reply:
Type for all `TIMESTAMP` fields modified for semantics.